airflow/api_connexion/parameters.py

```diff
@@ -86,3 +87,32 @@ def wrapped_function(*args, **kwargs):
         return cast(T, wrapped_function)
 
     return format_parameters_decorator
+
+
+def apply_sorting(model, query, order_by, to_replace=None, allowed_attrs=None):
+    """Apply sorting to query"""
+    lstriped_orderby = order_by.lstrip('-')
+    if allowed_attrs and lstriped_orderby not in allowed_attrs:
+        modelname = model.__tablename__.capitalize()
+        model_mapping = {
+            "Ab_user": 'User',
+            "Slot_pool": "Pool",
+            "Dag_run": "DagRun",
+            "Dag": "DagModel",
+            "Ab_role": "Role",
+            "Import_error": "ImportError",
+        }
+        if model_mapping.get(modelname, None):
+            modelname = model_mapping[modelname]
+        raise BadRequest(
+            detail=f"Ordering with '{lstriped_orderby}' is disallowed or "
+            f"the attribute does not exist on {modelname} model"
+        )
+    if to_replace:
+        for key, value in to_replace.items():
+            if key == order_by:
+                order_by = value
```

Comment:
```suggestion
        order_by = to_replace.get(order_by, order_by)
```

Also I feel we don’t need to repeat the ascending and descnding variants in the `to_replace` mapping. You can strip the `"-"` prefix before look up the `to_replace` mapping instead.

Something like

```python
if order_by[0] == "-":
    func = desc
    order_by = order_by[1:]
else:
    func = asc
if to_replace:  # This does not need to include `"-id"` etc. anymore, only `"id"`.
    order_by = to_replace.get(order_by, order_by)
return query.order_by(func(order_by))
```

Reply:
Awesome!